### PR TITLE
New json data source

### PR DESCRIPTION
**Pull request by susodapop**
Largely inspired by Arik's [gist](https://gist.github.com/arikfr/5a6b65dc63242abe1971d6b74ddd271d). Combined with the regular URL data source as there will be some crossover between these two.

@arikfr I'm particularly interested in your opinion of my decision to combine `URL` and `JSON` on the same page.

### CODE REVIEW

**Comment by ranbena** [comment]
@susodapop the images in this PR are relatively heavy. When help drawer opens it waits for page to fully load so should be optimized.
Please consider jpgs instead or run through PNG optimizer (tinypng.com will reduce 70%, quality intact).

**Comment by arikfr** [comment]
Is it possible to wait for everything but the images?

-- 
Arik Fraimovich

**Comment by ranbena** [comment]
> Is it possible to wait for everything but the images?

The only available iframe event listener is `load` which waits for all resources.
The only workaround is communicating with postMessage on `DOMContentLoaded`.
It's not too complicated but a bit of a hassle.

**Comment by arikfr** [comment]
Yeah, implementing a postMessage does feel too much. Why do we wait and not load it immediately? To avoid flickering of the header?

File: src/pages/kb/data-sources/querying-urls.md [hunk 1]
```diff
@@ -1,28 +1,56 @@
...
+toc: true
 ---
-Using a URL based data source requires that the URL use our format (see example below).
 
-The query itself inside Redash will simply contain the URL to be executed
-(i.e.   <http://myserver/path/myquery>) and result with all the data it has.
+Sometimes you need to visualize data not contained in an RDBMS or NOSQL data store. For those times, Redash provides two URL-based mechanisms to pull in your data: `URL` and `JSON`. Either source can be added from the settings screen in Redash.
 
-Here's a list of valid column types returned in results:
+`URL` is powerful because you can load large amounts of tabular data while abstracting away the underlying storage mechanism. This data source allows you to explicitly declare the types of data returned (text, dates, numbers e.g.). If a traditional database connection is unavailable, `URL` is the best way to load data to Redash.
```
Comment (arikfr) [inline src/pages/kb/data-sources/querying-urls.md]: The goal is to deprecate the `URL` data source. There is no point in keeping it except for legacy reasons.

Comment (susodapop) [reply src/pages/kb/data-sources/querying-urls.md]: I updated the body text and added a callout about upcoming deprecation.

**Comment by susodapop** [comment]
@ranbena I used ImageOptim to reduce the image size for the PNG's. I tried a JPEG as well but there was no space savings compared to the minified PNG.

**Comment by ranbena** [comment]
Btw I'm getting 40% more reduction with tinypng.com from the current images

**Pull request merged**